update-bot/wlmbots/lib/article_iterator.py

```diff
@@ -15,10 +15,8 @@ def __init__(self, callbacks, categories=None):
         self.articles_per_category_limit = 0
         self.log_every_n = 100
         self.callbacks = callbacks
-        if categories:
-            self.categories = categories
-        else:
-            self.categories = []
+        self.categories = [] if categories is None else categories
+        self._excluded_articles = {}
```

Comment:
Why is this a dictionary and not simply a list? I only see its keys being used in the further code. Is this some kind of performance issue (checking if key is in dictionary is probably faster than iterating through a list). What is a typical expected number of excluded articles then?